### ISSUE TITLE
Blueprints with url_prefix responding in HTTP 308

### ISSUE THREAD

**Issue by khera-shanu** [pallets/flask#3271]
### Expected Behavior
2xx response for an existing working URL

### Actual Behavior
blueprints with url_prefix started to respond with 308 in the latest flask (used pip)

### Environment
* Python version: Python 3.6.7
* Flask version: 1.0.3
* Werkzeug version: 0.15.4

**Comment by ThiefMaster** [comment]
Please provide a minimal piece of example code demonstrating this issue.

**Comment by khera-shanu** [comment]
read the issue, code won't be required.

**Comment by ThiefMaster** [comment]
Yes, code **is** required, because you seem to be the only one with this issue.

**Comment by khera-shanu** [comment]
https://github.com/khera-shanu/large/blob/master/app/views/user.py

this is the code

**Comment by ThiefMaster** [comment]
Again, please provide a MINIMAL example demonstrating your issue. That means, a single file/snippet one can run with `flask run`.

**Comment by XDfield** [comment]
same here.
@khera-shanu 
try setting `strict_slashes` to False

```python
app.url_map.strict_slashes = False
```

**Comment by saltaverde** [comment]
@khera-shanu I ran into this problem when refactoring all of my app's routes into separate `Blueprint` objects. There seems to be a difference in the defaults of `Flask.route` and `Blueprint.route`. The former worked as (I) expected with respect to trailing slashes in the url.

For example
```
app = Flask(__name__)

@app.route('/user')
def user():
    . . .
```
would work the same whether the request was to `/user` or `/user/`.

However, in order to make a `Blueprint` behave this way, I had to add `strict_slashes=False` to each of the `Blueprint.route` calls.

For example
```
bp = Blueprint('user', __name__, url_prefix='/user')

@bp.route('/', strict_slashes=False)
def user():
    . . .
```

**Comment by qihboy** [comment]
same issue found, and ```app.url_map.strict_slashes = False``` not worked for us.

We found if blueprint.route registered route for sub modules, like ```/api/module1/``` for ```app/module/entrance.py``` and defined route like ``` route = ['/', '']``` in entrance.py, 
when request accepted, it will return a 308 response and then a 200 response, some clients like web browser will handle this, some others will not(such like another web service).

We fixed this by redefine route for this situation:``` route = '/callbak'```, and now it's working for good.
